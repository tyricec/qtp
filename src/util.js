const util = {
  createElement: function(type) {
    const element = document.createElement(type)
    return element
  },
}

export default util