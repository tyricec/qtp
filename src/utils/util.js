const util = {
  buildHTML: (type, attrs, inner) => {
    let element

    if (type) {
      element = document.createElement(type)
    }
    
    if (element && attrs) {
      Object.keys(attrs).forEach((attr) => {
        element.setAttribute(attr, attrs[attr])
      })
    }

    if (element && inner) {
      element.appendChild(inner)
    }

    return element
  },
  buildText: (text) => {
    return document.createTextNode(text)
  },
}

export default util