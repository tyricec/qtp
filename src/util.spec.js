import util from './util'

test('createElement creates div element', () => {
  const result = util.createElement('div')

  expect(result.outerHTML).toMatchSnapshot()
})

test('createElement create li element', () => {
  const result = util.createElement('li')

  expect(result.outerHTML).toMatchSnapshot()
})