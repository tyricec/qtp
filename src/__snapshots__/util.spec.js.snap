// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement create li element 1`] = `"<li></li>"`;

exports[`createElement creates div element 1`] = `"<div></div>"`;
